### DISCUSSION TITLE
利用WebAV合并下载的视频音频流

### DISCUSSION THREAD

**Discussion by oxygenkun** [the1812/Bilibili-Evolved#4615] [功能建议 / Ideas]
`v2.8.6`开始提供了基于`ffmpeg.wasm`实现音视频合并下载输出。但是加载lib需要一定的等待时间。

[`WebAV`](https://github.com/bilibili/WebAV)应该也能实现类似的功能，并且更加轻量。

> [WebAV音视频合并DEMO](https://bilibili.github.io/WebAV/demo/2_2-video-add-audio)

**Comment by oxygenkun** [comment]
还有一些同样基于 `WebCodecs API` 实现混流的项目：
- [webm-muxer](https://github.com/Vanilagy/webm-muxer)
- [mp4-muxer](https://github.com/Vanilagy/mp4-muxer)

**Comment by WakelessSloth56** [comment]
我当时写 `ffmpeg.wasm` 插件的时候就有想到用 `WebCodecs`，但是考虑到这个东西兼容性不好，比如 Firefox 就根本不支持，所以还是用了 FFmpeg。https://caniuse.com/webcodecs

我之后再写一个基于 `WebCodecs` 的吧。
